$OpenBSD: patch-lib_Makefile,v 1.9 2020/09/03 17:11:10 bket Exp $

- Some compilers don't have -Wvla
- Don't set compiler optimisation level
- Don't create symbolic links to SHARED_LIB

Index: lib/Makefile
--- lib/Makefile.orig
+++ lib/Makefile
@@ -38,7 +38,7 @@ endif
 DEBUGFLAGS= -Wall -Wextra -Wcast-qual -Wcast-align -Wshadow \
             -Wstrict-aliasing=1 -Wswitch-enum -Wdeclaration-after-statement \
             -Wstrict-prototypes -Wundef -Wpointer-arith \
-            -Wvla -Wformat=2 -Winit-self -Wfloat-equal -Wwrite-strings \
+            -Wformat=2 -Winit-self -Wfloat-equal -Wwrite-strings \
             -Wredundant-decls -Wmissing-prototypes -Wc++-compat
 CFLAGS  += $(DEBUGFLAGS) $(MOREFLAGS)
 FLAGS    = $(CPPFLAGS) $(CFLAGS)
@@ -81,7 +81,7 @@ ifneq ($(ZSTD_LIB_MINIFY), 0)
 	CFLAGS += -fno-stack-protector -fomit-frame-pointer -fno-ident \
 	          -DDYNAMIC_BMI2=0 -DNDEBUG
 else
-	CFLAGS += -O3
+	CFLAGS +=
 endif
 
 # Modules
@@ -328,8 +328,6 @@ install-shared: libzstd
 	@echo Installing shared library
 	$(Q)$(INSTALL) -d -m 755 $(DESTDIR)$(LIBDIR)/
 	$(Q)$(INSTALL_PROGRAM) $(LIBZSTD) $(DESTDIR)$(LIBDIR)
-	$(Q)ln -sf $(LIBZSTD) $(DESTDIR)$(LIBDIR)/libzstd.$(SHARED_EXT_MAJOR)
-	$(Q)ln -sf $(LIBZSTD) $(DESTDIR)$(LIBDIR)/libzstd.$(SHARED_EXT)
 
 install-includes:
 	@echo Installing includes
